src/analyses/threadEscape.ml

```diff
@@ -63,10 +63,10 @@ struct
     if M.tracing then M.tracel "escape" "assign vs: %a\n" D.pretty vs;
     if D.exists (fun v -> v.vglob || has_escaped ask v) vs then (
       let escaped = reachable ask rval in
-      let escaped = D.filter (fun v -> not v.vglob) escaped in
+      let escaped = D.filter (fun v -> not v.vglob || ctx.ask (IsHeapVar v)) escaped in
```

Comment:
The potential thing to be extracted from this PR before closing is this (and similar changes below). It also considers heap variables to be escaped, besides just locals.
It's not clear whether this was some soundness fix, without a corresponding test, or something specific to this PR. Maybe it was specific to other changes which are about heap memory.

Also, since this, there's now an `IsAllocVar` query, which might be more appropriate if this still has any purpose.

Reply:
We thought about it for a while and concluded that such extra query is not necessary to fix anything on `master`.
It was introduced in a4ac9c765b44111c2f739471e44d694c28fcf345 only because the `event` handler added to the new mallocFresh cared about alloc variables having escaped. Since the current version doesn't care about the event at all, this makes no difference.
The escape analysis itself already behaves correctly with allocated memory and we have regression tests for it.

Reply:
I guess this means the PR can be closed?

Reply:
Yup, @karoliineh will still add `combine_assign` to the escape analysis, but that's a separate PR, nothing to reuse from here.